python/tests/test_transform.py

```diff
@@ -1,15 +1,126 @@
+import datetime
 import pandas as pd
 import numpy as np
 import pymongo
+import pytest
 from energuide import database
 from energuide import transform
 
+@pytest.fixture
+def sample() -> pd.DataFrame:
+    data = {
+        'EVAL_ID': [123456],
+        'IDNUMBER': [23],
+        'CREATIONDATE': ['2009-01-01 12:01:02'],
+        'MODIFICATIONDATE': ['2011-01-01 00:01:02'],
+        'YEARBUILT': [1979],
+        'HOUSEREGION': ['Ontario'],
+        'CLIENTCITY': ['Kingston'],
+        'CLIENTPCODE': ['K0H 1Y0'],
+    }
+    return pd.DataFrame.from_dict(data)
 
-def test_clean() -> None:
-    dataframe = pd.DataFrame.from_dict({'foo': [np.NaN]}, orient='columns')
-    output = transform.clean(dataframe).to_dict('records')
+@pytest.fixture()
+def renamed_sample(sample: pd.DataFrame) -> pd.DataFrame:
+    return transform.rename_columns(sample)
 
-    expected = [{'foo': None}]
+
+def test_clear_blanks() -> None:
+    dataframe = pd.DataFrame.from_dict({'foo': [np.NaN], 'bar': ['some']}, orient='columns')
+    output = transform.clear_blanks(dataframe).to_dict('records')
+
+    expected = [{'foo': None, 'bar': 'some'}]
+    assert output == expected
+
+
+def test_rename_columns(sample: pd.DataFrame)  -> None:
+    dataframe = transform.rename_columns(sample)
+    expected = [
+        'clientCity',
+        'clientPostalCode',
+        'creationDate',
+        'evalId',
+        'houseRegion',
+        'idNumber',
+        'modificationDate',
+        'yearBuilt'
+    ]
+
+    assert sorted(list(dataframe.columns)) == expected
+
+
+def test_extract_postal(renamed_sample: pd.DataFrame)  -> None:
+    output = transform.extract_postal(renamed_sample).to_dict('records')
+
+    expected = [{
+        'evalId': 123456,
+        'idNumber': 23,
+        'yearBuilt': 1979,
+        'houseRegion': 'Ontario',
+        'clientCity': 'Kingston',
+        'clientForwardSortationArea': 'K0H',
+        'clientPostalCode': 'K0H 1Y0',
+        'creationDate': '2009-01-01 12:01:02',
+        'modificationDate': '2011-01-01 00:01:02'
+    }]
+
+    assert output == expected
+
+
+def test_parse_dates(renamed_sample: pd.DataFrame)  -> None:
+    output = transform.parse_dates(renamed_sample).to_dict('records')
+
+    expected = [{
+        'evalId': 123456,
+        'idNumber': 23,
+        'yearBuilt': 1979,
+        'houseRegion': 'Ontario',
+        'clientCity': 'Kingston',
+        'clientPostalCode': 'K0H 1Y0',
```

Comment:
If these fields are not necessary to test the `parse_dates` function, let's remove them.